p9-presentation-methodology (#75)
chhoumann/thesis-chemcam: p9-presentation/sections/_methodology.qmd

That means we create predictions for each of 8 major oxides.<br>
However, most of the steps actually involve data being processed for each oxide individually.<br>
For both ICA and PLS, you can imagine that after some preprocessing steps, we branch out into 8 branches, each of which predict its own values for its associated oxide.<br>
And in the end, we put the outputs together to produce the MOC prediction.<br>
<br>
I start by walking through the PLS-SM side, and then the ICA side.
:::

<!-- ## PLS-SM 
- Recreating the PLS-SM method presented by @cleggRecalibrationMarsScience2017 and @andersonImprovedAccuracyQuantitative2017
    - Preprocessing
    - Outlier removal
    - Training
    - Inference via submodels

::: {.notes}
Starting with the PLS-SM method, we have attempted to recreate the system presented by both Clegg and Anderson et al.
::: -->

## PLS-SM Motivation {.unlisted auto-animate="true"}
- **PLS**
    - Good for chemometrics
    - Readily intrepretable

::: {.notes}
Let me start by explaining what motivated PLS-SM, because that will help us understand why it does what it does.<br>
<br>

> First: they chose PLS because it can handle noisy data with many more variables than observations, and with significant correlation between variables.<br>

Reviewer: ```suggestion
First: they chose PLS because it can handle noisy data with far more variables than observations, and with significant correlation between variables.<br>
```

It's also readily intrepretable by plotting the regression coefficients as a function of the spectral channels. This shows which wavelength ranges show correlations with the prediction composition.
:::

## PLS-SM Motivation {.unlisted auto-animate="true"}
- PLS
- **PLS2 vs. PLS1**
    - PLS2: Multivariate
    - PLS1: Univariate
    - Better results with separate PLS1 models

::: {.notes}
For the original ChemCam calibration, the team used PLS2. This case corresponds to the case where there are several dependent variables, meaning it can perform multivariate predictions.<br>
<br>

> However, they found that they're seeing better results by using a separate PLS1 model for each of the major oxides.<br>

Reviewer: ```suggestion
However, they achieved better results by using a separate PLS1 model for each of the major oxides.<br>
```

| K2O   | (0, 100) | (0, 2)  | *N/A*    | (1.5, 100)|

::: {.footer}
All submodel PLS configurations are taken from @cleggRecalibrationMarsScience2017.
:::

::: {.notes}
These are the compositional ranges that @cleggRecalibrationMarsScience2017 presented in their paper.<br>
<br>
As can be seen, the ranges overlap.<br>
This is to avoid discontinuities in the final combined results.<br>
They would occur because the samples in each sub-model with the most "extreme" compositions aren't necessarily weighted as strongly when the model is trained, so may be less accurately predicted.<br>
<br>
The N/As here mean that there are no compositional ranges defined there.<br>
Essentially meaning there are no Mid submodels for sodium and potassium oxide.
:::

## PLS-SM Inference {.unlisted}
![](/static/methodology/pls_inference.png){fig-align="center" width="70%"}

::: {.notes}
Let me illustrate.<br>
<br>
<!-- I'm starting out by showing you how the model makes predictions, because that is the simplest view to start from.<br>
<br> -->
It's important to understand that, in practice, this flow is executed for each oxide.<br>
For simplicity's sake, we present it as being done wholly.<br>
<br>

> The full model is used to estimate the composition of an unknown target, and the appropriate sub-model(s) is then chosen based on this estimation for a more accurate prediction.<br>

Reviewer: ```suggestion
The full model is used to estimate the composition of an unknown target, and the appropriate sub-model(s) is then chosen based on this estimate for a more accurate prediction.<br>
```

So for each oxide, we filter it by the compositional range associated with a given submodel.<br>
This gives us the training data for this model's compositional range.
:::

## Preprocessing {auto-animate="true" auto-animate-id="p1" transition="slide-in fade-out" .unlisted .smaller}
:::: {.columns}
::: {.column width="50%"}
- Remove 'dust' shots
- Average shot intensities
- Mask noisy wavelenghts
- Zero out negative values
- Tranpose
- Submodels filter
- Normalize
    - Norm 1: *normalize full spectrum,* $\text{sum}=1$
    - Norm 3: *normalize per spectrometer,* $\text{sum}=3$
:::

::: {.column width="50%"}
![PLS Preprocessing](static/methodology/pls_preprocessing.png){style="position: relative; top:-1200px;" data-id="x1"}
<div style="position: absolute; top: 25%; left: 55%; width: 38%; height: 45%; border: 2px solid red; box-sizing: border-box; pointer-events: none; border-radius: 10px;" data-id="pbox"></div>
:::
::::

::: {.notes}
Now we normalize.<br>
<br>
**Norm 1** normalizes full spectrum by total across the total across all three spectrometers, so the resulting spectrum adds up to 1.<br>
<br>

> **Norm 3** normalizes on a per-spectrometer basis, resulting in a full normalized spectrum adds up to 3 (because there are 3 separate detectors...).

Author: Yeah

<br>
As you can see, those items with both a high error and leverage are classified as outliers and are promptly removed.
:::

## MOC Pipeline {auto-animate="true"}
::: {style="width=100%;"}
![](/static/methodology/pipeline.png){fig-align="center" width="35%"}
:::

::: {data-id="moc-box"}
<div style="position: absolute; top: 34.45%; left: 51.25%; width: 18%; height: 35.25%; border: 2px solid red; box-sizing: border-box; pointer-events: none; border-radius: 10px;"></div>
:::

## ICA Preprocessing
:::: {.columns}
::: {.column width="50%"}
- Similar preprocessing to PLS-SM, except:
    - No averaging
    - No compositional range filtering
    - Uses only one location per sample
:::
::: {.column width="50%"}
![ICA Preprocessing](static/methodology/ica_preprocessing.png){width="55%"}
:::
::::

::: {.notes}
I won't go into much detail about the preprocessing stage in the ICA phase.<br>
The process mostly follows the same steps as in PLS-SM, except we don't filter for any compositional ranges here.<br>
<br>
We also do not average the shots, and instead preserve the intensity values for each shot, except the first five.<br>
<br>

> Finally, we don't use all five location datasets for each sample. That is left as an experiment, which Ivik will go into details about. We chose to do it this way because it was unclear whether the original authors used all location datasets or just one per sample.

Reviewer: ```suggestion
Finally, we don't use all five location datasets for each sample. That is left as an experiment, which Ivik will go into detail about. We chose to do it this way because it was unclear how the original authors used all location datasets.
```
IMO it was clear they used all five - but it wasn't clear how they did it. That's also what we said in our report.
